yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -209,7 +208,7 @@ export class Archiver implements L2BlockSource, L2LogsSource, ContractDataSource
     // ********** Events that are processed per block **********
 
     // Read all data from chain and then write to our stores at the end
-    const nextExpectedL2BlockNum = BigInt(this.store.getBlocksLength() + INITIAL_L2_BLOCK_NUM);
+    const nextExpectedL2BlockNum = BigInt((await this.store.getBlockNumber()) + 1);
```

Comment:
Doesn't this break if we restart the archiver? Say L2 is at block 30. Say archiver restarts. So `nextExpectedL2BlockNum` would be 1. But I suppose we don't care about re-syncing until testnet?

Reply:
It would have been one in the first instance too?

Reply:
This seems correct to me because when we start archiver the next expected L2 block num is 1 becuase that's the first block.